packages/aws-cdk-lib/aws-ecr/test/repository.test.ts

```diff
@@ -1373,3 +1373,234 @@ describe('repository', () => {
     });
   });
 });
+
+describe('LifecycleRuleClass', () => {
```

Comment:
The original issues asked for use with `CfnRepositoryCreationTemplate`, but the tests don't show that.